cmd/grafana-app-sdk/project.go

```diff
@@ -609,34 +611,109 @@ func projectAddPluginAPI[G anyGenerator](generator G, repo, generatedAPIModelsPa
+	if !isCommandInstalled("yarn") {
+		return fmt.Errorf("yarn must be installed to add the frontend component")
+	}
+
+	args := []string{"create", "@grafana/plugin", "--pluginType=app", "--hasBackend=true", "--pluginName=tmp", "--orgName=tmp"}
+	cmd := exec.Command("yarn", args...)
+	buf := bytes.Buffer{}
+	ebuf := bytes.Buffer{}
+	cmd.Stdout = &buf
+	cmd.Stderr = &ebuf
+	err := cmd.Start()
 	if err != nil {
 		return err
 	}
-	err = writePluginJSON(filepath.Join(projectRootPath, "plugin/src/plugin.json"),
-		fmt.Sprintf("%s-app", pluginID), "NAME", "AUTHOR", pluginID)
+	fmt.Println("Creating plugin frontend using `\033[0;32myarn create @grafana/plugin\033[0m` (this may take a moment)...")
+	err = cmd.Wait()
 	if err != nil {
+		// Only print command output on error
+		fmt.Println(buf.String())
+		fmt.Println(ebuf.String())
 		return err
 	}
-	err = writePluginConstants(filepath.Join(projectRootPath, "plugin/src/constants.ts"), pluginID)
+
+	// Remove a few directories that get created which we don't actually want
+	err = os.RemoveAll("./tmp-tmp-app/.github")
 	if err != nil {
 		return err
 	}
-	err = writePackageJSON(filepath.Join(projectRootPath, "plugin/package.json"), "NAME", "AUTHOR")
+	err = os.RemoveAll("./tmp-tmp-app/pkg")
 	if err != nil {
 		return err
 	}
-	return nil
+	err = os.Remove("./tmp-tmp-app/go.mod")
+	if err != nil {
+		return err
+	}
+	err = os.Remove("./tmp-tmp-app/go.sum")
+	if err != nil {
+		return err
+	}
+	// Move the remaining contents into /plugin
+	err = moveFiles("./tmp-tmp-app/", filepath.Join(projectRootPath, "plugin"))
+	if err != nil {
+		return err
+	}
+	err = writePluginJSON(filepath.Join(projectRootPath, "plugin/src/plugin.json"),
+		fmt.Sprintf("%s-app", pluginID), "NAME", "AUTHOR", pluginID)
+	if err != nil {
+		return err
+	}
+	err = writePluginConstants(filepath.Join(projectRootPath, "plugin/src/constants.ts"), pluginID)
+	if err != nil {
+		return err
+	}
+	return os.Remove("./tmp-tmp-app")
+}
+
+func moveFiles(srcDir, destDir string) error {
+	return filepath.WalkDir(srcDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Just move directories wholesale by renaming
+		if d.IsDir() {
+			if path == srcDir {
+				return nil
+			}
+			dst := filepath.Join(destDir, d.Name())
+			if _, serr := os.Stat(dst); serr == nil {
+				err := moveFiles(path, dst)
+				if err != nil {
+					return err
+				}
+				if err = os.Remove(path); err != nil {
+					return err
+				}
+				return fs.SkipDir
+			}
+			err = os.Rename(path, filepath.Join(destDir, d.Name()))
+			if err != nil {
+				return err
+			}
+			return fs.SkipDir
+		}
+
+		return os.Rename(path, filepath.Join(destDir, d.Name()))
+	})
+}
+
+func isCommandInstalled(command string) bool {
+	cmd := exec.Command("which", command)
+	err := cmd.Run()
+	return err == nil
 }
 
 func writePluginJSON(fullPath, id, name, author, slug string) error {
```

Comment:
Could we also stop using templated `plugin.json` as well and instead read the one we have generated and overwrite specific fields we want to tweak?

We could even use https://github.com/grafana/grafana/blob/main/pkg/plugins/plugins.go#L137-L191, possibly.

Reply:
Yup, I'd like to update this behavior as well, but I think I'd like to have that be a future PR for the same reason as the other comment--for this PR, I'm trying to not change too much in one go, and leave the current behavior more intact.